pages/pil/unscoped/courses/routers/update.js

```diff
@@ -0,0 +1,42 @@
+const { Router } = require('express');
+const moment = require('moment');
+const { form } = require('../../../../common/routers');
+const schema = require('../schema/update');
+
+module.exports = () => {
+  const app = Router({ mergeParams: true });
+
+  app.use(form({
+    schema,
+    getValues: (req, res, next) => {
+      const projectId = req.form.values.projectId;
+      if (!projectId) {
+        return next();
+      }
+      req.api(`/establishment/${req.establishmentId}/projects/${projectId}`)
+        .then(response => response.json.data)
+        .then(project => {
+          req.form.schema.projectId.defaultValue = project.licenceNumber;
+          next();
+        })
+        .catch(next);
+    },
+    process: (req, res, next) => {
+      const day = req.body['startDate-day'];
+      const month = req.body['startDate-month'];
+      const year = req.body['startDate-year'];
+
+      Object.assign(req.form.values, {
+        startDate: `${year}-${month}-${day}`
+      });
+
+      next();
+    },
+    saveValues: (req, res, next) => {
+      req.session.form[req.model.id].values.startDate = moment(req.form.values.startDate, 'YYYY-MM-DD').format('YYYY-MM-DD');
```

Comment:
What is the distinction between this and the broadly equivalent processing being done in `process` above? How come it's split in two?

Reply:
pass, I lifted this from somewhere else

Reply:
I think process happens before validation, and save values happens after off the top of my head

Reply:
That makes sense. Thanks.